From f6f9ab4d88e37ff37083f8c5b44dd04e6a395190 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 16:43:52 +0800
Subject: [PORT FROM MAIN] Fix the fps less than 30 issue

BZ: 50971

refine the call sequence of the fillOutputBuffers and drainInputBuffers to
enhance the fps performance

Change-Id: I1726159e6b6133d29519e28578f25b8a09cfc6e5
Orig-Change-Id: Ib70fa1f29f7b793da647cf81768fee323203c0d2
Signed-off-by: jiguoliang <guoliang.ji@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/OMXCodec.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 24900ea..0a07a59 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -3985,13 +3985,13 @@ status_t OMXCodec::read(
             mPaused = false;
         }
 
-        drainInputBuffers();
 
         if (mState == EXECUTING) {
             // Otherwise mState == RECONFIGURING and this code will trigger
             // after the output port is reenabled.
             fillOutputBuffers();
         }
+        drainInputBuffers();
     }
 
     if (seeking) {
-- 
1.7.4.1

